pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -261,12 +274,12 @@ func (asw *actualStateOfWorld) AddVolumeToReportAsAttached(
 }
 
 func (asw *actualStateOfWorld) AddVolumeNode(
-	uniqueName v1.UniqueVolumeName, volumeSpec *volume.Spec, nodeName types.NodeName, devicePath string) (v1.UniqueVolumeName, error) {
+	uniqueName v1.UniqueVolumeName, volumeSpec *volume.Spec, nodeName types.NodeName, devicePath string, isAttached bool) (v1.UniqueVolumeName, error) {
 	asw.Lock()
 	defer asw.Unlock()
 
-	var volumeName v1.UniqueVolumeName
-	if volumeSpec != nil {
+	volumeName := uniqueName
+	if volumeName == "" {
```

Comment:
why did we remove the nil check for `volumeSpec` here? it is still used throughout this function

Reply:
So when calling this function, the caller should make sure at least specify either volumeName (not empty) or volumeSpec (not nil). The called should make sure not specify both empty volumeName and nil volumeSpec. The logic here I changed to is
if volumeName is not empty, use volumeName. Otherwise, use volumeSpec. 

Before, it tries to use volumeSpec first, and then volumeName.

I will add the nil check for safety.